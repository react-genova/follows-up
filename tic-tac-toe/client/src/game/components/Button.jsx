import React from 'react';
import PropTypes from 'prop-types';
import StyledButton from './Button.styled';

const Button = ({ text, onClick }) => (
    <StyledButton type="button" onClick={onClick}>{text}</StyledButton>
);

Button.propTypes = {
    text: PropTypes.string,
    onClick: PropTypes.func,
};

Button.defaultProps = {
    text: '',
    onClick: undefined,
};

export default Button;
