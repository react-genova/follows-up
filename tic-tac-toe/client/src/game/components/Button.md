### Cancel button

```jsx
<div style={{ backgroundColor: '#333', padding: '1em' }}>
    <Button text="Cancel" />
</div>
```

### Play again button

```jsx
<div style={{ backgroundColor: '#333', padding: '1em' }}>
    <Button text="Play again" />
</div>
```
