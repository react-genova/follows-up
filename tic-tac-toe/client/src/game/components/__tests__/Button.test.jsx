import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import Button from '../Button';

describe('Button component', () => {
    afterEach(cleanup);

    it('renders without crashing (empty)', () => {
        const { container } = render(<Button />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders without crashing (filled)', () => {
        const { container } = render(
            <Button text="Play again" />,
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
