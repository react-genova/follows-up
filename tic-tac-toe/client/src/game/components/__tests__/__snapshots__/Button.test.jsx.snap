// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders without crashing (empty) 1`] = `
<button
  class="sc-bdVaJa kJcqmF"
  type="button"
/>
`;

exports[`Button component renders without crashing (filled) 1`] = `
<button
  class="sc-bdVaJa kJcqmF"
  type="button"
>
  Play again
</button>
`;
